vg/recorder/recorder.go

```diff
@@ -381,29 +384,69 @@ type FillString struct {
 }
 
 // FillString implements the FillString method of the vg.Canvas interface.
-func (c *Canvas) FillString(font vg.Font, x, y vg.Length, str string) {
+func (c *Canvas) FillString(font vg.Font, pt vg.Point, str string) {
 	c.append(&FillString{
-		Font: font.Name(),
-		Size: font.Size,
-		X:    x, Y: y,
+		Font:   font.Name(),
+		Size:   font.Size,
+		Point:  pt,
 		String: str,
 	})
 }
 
 // ApplyTo applies the action to the given vg.Canvas.
 func (a *FillString) ApplyTo(c vg.Canvas) {
-	c.FillString(a.fonts[fontID{name: a.Font, size: a.Size}], a.X, a.Y, a.String)
+	c.FillString(a.fonts[fontID{name: a.Font, size: a.Size}], a.Point, a.String)
 }
 
 // Call returns the pseudo method call that generated the action.
 func (a *FillString) Call() string {
-	return fmt.Sprintf("%sFillString(%q, %v, %v, %v, %q)", a.l, a.Font, a.Size, a.X, a.Y, a.String)
+	return fmt.Sprintf("%sFillString(%q, %v, %v, %v, %q)", a.l, a.Font, a.Size, a.Point.X, a.Point.Y, a.String)
 }
 
 func (a *FillString) callerLocation() *callerLocation {
 	return &a.l
 }
 
+// DrawImage corresponds to the vg.Canvas.DrawImage method
+type DrawImage struct {
+	Rectangle vg.Rectangle
+	Image     []byte
+
+	l callerLocation
+}
+
+// DrawImage implements the DrawImage method of the vg.Canvas interface.
+func (c *Canvas) DrawImage(rect vg.Rectangle, img image.Image) {
+	buf := new(bytes.Buffer)
+	err := png.Encode(buf, img)
+	if err != nil {
+		panic(fmt.Errorf("recorder: error encoding image to PNG: %v\n", err))
+	}
+	c.append(&DrawImage{
+		Rectangle: rect,
+		Image:     buf.Bytes(),
+	})
+}
+
+// ApplyTo applies the action to the given vg.Canvas.
+func (a *DrawImage) ApplyTo(c vg.Canvas) {
+	buf := bytes.NewReader(a.Image)
+	img, err := png.Decode(buf)
+	if err != nil {
+		panic(fmt.Errorf("recorder: error decoding image from PNG: %v\n", err))
+	}
+	c.DrawImage(a.Rectangle, img)
+}
+
+// Call returns the pseudo method call that generated the action.
+func (a *DrawImage) Call() string {
+	return fmt.Sprintf("%sDrawImage(%v, %v)", a.l, a.Rectangle, a.Image)
```

Comment:
Here, if we retain the image as an image.Image, either render the call as just the bounds of the image or as that plus the tour/pic format ("IMAGE:"+bas64EncodingOfPNG).